staramr/results/AMRDetectionSummary.py

```diff
@@ -58,4 +70,12 @@ def create_summary(self, include_negatives=False):
 
         df.rename(columns={'Gene': 'Genotype'}, inplace=True)
 
+        if ds is not None:
+            ds = self._compile_plasmids(ds)
+
+            if not ds.empty and not df.empty:
+                df = df.merge(ds, on='Isolate ID', how='left').drop(['Plasmid Genes_x'], axis=1)
```

Comment:
So, I'm wondering if you can try something out for me?

Instead of adding a **Plasmid Genes** column in the `_include_negatives` method (which then results in these **Plasmid Genes_x/y** columns that you have to drop), let's just do a `left` join on these two tables.  Then, you can do `fillna(values={'Plasmid Genes': 'None'})`, which should fill all the empty **Plasmid Genes** columns with `None`).

That is:

```python
df = df.merge(ds, on='Isolate ID', how='left').fillna(values={'Plasmid Genes': 'None'})
```

And then remove changes you've made in `AMRDetectionSummaryResistance._include_negatives()`. This also means you don't have to have code dropping column **Plasmid Genes_x/y** (there should only be one **Plasmid Genes** column now).

I think this should work, but haven't tested it out. Could you try it out for me?

Reply:
Thanks so much for looking into this. I'm glad it all worked out :smile: